lib/route.js

```diff
@@ -327,7 +327,8 @@ function buildRouting (options) {
               schemaController.setupValidator(this[kOptions])
             }
             try {
-              compileSchemasForValidation(context, opts.validatorCompiler || schemaController.validatorCompiler)
+              const isCustom = typeof opts?.validatorCompiler === 'function' || schemaController.customValidatorCompiler
+              compileSchemasForValidation(context, opts.validatorCompiler || schemaController.validatorCompiler, isCustom)
```

Comment:
Should we do the same for `compileSchemasForSerialization` for consistency?

Reply:
I have thought about it before. But serialization didn't mutate the schema and I would leave it for now.

The `isCustomSerializerCompiler` is added only because it maybe useful in future.